clang/test/Driver/sycl-obj-remove.cpp

```diff
@@ -0,0 +1,26 @@
+/// Verify object removal when the offload compilation fails.
+
+// REQUIRES: system-linux
+
+/// Force failure during device compilation
+// RUN: touch %t.o
+// RUN: not %clangxx -fsycl -Xsycl-target-frontend -DCOMPILE_FAIL=1 -c -o %t.o %s
+// RUN: not ls %t.o
```

Comment:
Could you explain how running the `ls` command can confirm that the object file has been successfully removed after a compilation failure?

If we try to `ls` a file that is not present, we would get
ls: cannot access 't.o': No such file or directory

Can you check for that message instead?

Reply:
`ls` as a tool will return a `0` or a `1` depending on if the command was successful.  So running `ls <file>` will return a `0` if the file was found.  If the file is not found, it will return a `1`.  So doing a `not ls` will allow for error codes to show pass or fail within the LIT testing infrastructure.